xchange-huobi/src/main/java/org/knowm/xchange/huobi/HuobiAdapters.java

```diff
@@ -92,19 +88,54 @@ static ExchangeMetaData adaptToExchangeMetaData(
       pairs.put(pair, adaptPair(assetPair, pairsMetaData.getOrDefault(pair, null)));
     }
 
-    Map<Currency, CurrencyMetaData> currenciesMetaData = staticMetaData.getCurrencies();
     Map<Currency, CurrencyMetaData> currencies = new HashMap<>();
-    for (HuobiAsset asset : assets) {
-      Currency currency = adaptCurrency(asset.getAsset());
-      CurrencyMetaData metadata = currenciesMetaData.getOrDefault(currency, null);
-      BigDecimal withdrawalFee = metadata == null ? null : metadata.getWithdrawalFee();
-      int scale = metadata == null ? 8 : metadata.getScale();
-      currencies.put(currency, new CurrencyMetaData(scale, withdrawalFee));
+    for (HuobiCurrencyWrapper huobiCurrencyWrapper : currencyWrapper) {
+      if(huobiCurrencyWrapper.getHuobiCurrencies().length != 0) {
+        boolean isDelisted = DELISTED.equals(huobiCurrencyWrapper.getInstStatus());
+        CurrencyMetaData currencyMetaData = adaptCurrencyMetaData(huobiCurrencyWrapper, isDelisted);
+        Currency currency = HuobiUtils.translateHuobiCurrencyCode(huobiCurrencyWrapper.getCurrency());
+        currencies.put(currency, currencyMetaData);
+      }
     }
 
     return new ExchangeMetaData(pairs, currencies, null, null, false);
   }
 
+  private static CurrencyMetaData adaptCurrencyMetaData(HuobiCurrencyWrapper huobiCurrencyWrapper, boolean isDelisted) {
+    CurrencyMetaData result = null;
+    List<HuobiCurrency> huobiCurrencies = Arrays.asList(huobiCurrencyWrapper.getHuobiCurrencies());
+    if(!huobiCurrencies.isEmpty()) {
+      result = getCurrencyMetaData(huobiCurrencies.get(0), isDelisted);
+      for (HuobiCurrency huobiCurrency : huobiCurrencies) {
```

Comment:
this looks a little inefficient - it would be better to use a Map instead of this method for each currency

Reply:
@walec51 Huobi network response was in form of array. If we want to make it map It will require deserialization impl. for HuobiCurrenciesResult which will not improve our efficency much since metadata information is where currencies information is held and this list will be useless after we will metadata information.

Reply:
@walec51 Can you further explain the issue here ? I think your comment is misunderstood.

Reply:
it not that important, skip this comment